presentation/frontend/src/presentation-frontend/PresentationManager.ts

```diff
@@ -580,38 +585,60 @@ export class PresentationManager implements IDisposable {
     });
 
     return {
-      ...(await generator.createAsyncIteratorResponse()),
+      ...(await generator.createObservableResponse()),
       descriptor,
     };
   }
 
-  /** Retrieves a content descriptor, item count and async generator for the items themselves. */
-  public async getContentIterator(
+  /** @internal */
+  public async getContentObservable(
```

Comment:
Yes, I just don't feel like this workflow is efficient or neccessary: 
1. We convert the observable to an iterator in `PresentationManager`
2. In `HiliteSetProvider` we convert it back to observable so that we could pipe it to `shareReplay`
3. Then we convert it back to an iterator so that we could return it to the consumer.

With `getContentObservable` we could eliminate the first and second conversions

Reply:
Agree. What I don't like about having a public method returning an `Observable`, though, is that it adds `rxjs` to our API summary. From the API correctness standpoint that's not a problem, because the method is `@internal`, however it makes it easier to miss `rxjs` APIs being exposed through our public API (which we strictly don't want) in the future.

Maybe the method could be moved out of `PresentationManager` class as a standalone function instead?

Reply:
What about `ensureIModelInitialized` that we have to call before returning the result `Item`? Also, if we moved it to a function it would still have to be in `PresentationManager.ts` and marked `export`, so I don't see how it would help. I think the easiest solution would be to stick with `getContentIterator`, although I don't know how it would affect the performance.

Reply:
> What about `this.ensureIModelInitialized` that we have to call before returning the result?

We only need to await on it when the result contains a descriptor (in case caller wants to check if one of the descriptor  fields is favorited by a user). For hilite sets we don't care about that, if that's you're worried about.

> Also, if we moved it to a function it would still have to be in `PresentationManager.ts` and marked `export`, so I don't see how it would help.

Right... Unless the function is some other file that's not exported through the barrel :)